src/olympia/promoted/models.py

```diff
@@ -117,21 +117,37 @@ def get_resigned_version_number(self):
         else:
             return None
 
+    def _set_needs_human_review_on_version(self, version, due_date):
+        had_due_date_already = bool(version and version.due_date)
+        if (
+            not version
+            or version.needs_human_review
+            or self._get_approved_applications_for_version(version)
+        ):
+            return
+        version.update(needs_human_review=True)
+        if not had_due_date_already and due_date:
+            # If we have a specific due_date, override the default
+            version.reset_due_date(due_date)
+
     def save(self, *args, **kwargs):
```

Comment:
This PR is still blocked, but note that when unblocked it'll need to integrate the changes made for https://github.com/mozilla/addons-server/issues/20405

Reply:
Changes made during the rebase